sherpa/models/tests/test_template.py

```diff
@@ -64,35 +63,44 @@ def test_restore_is_discrete(self):
     # TestCase 1 load_template_model enables interpolation by default
     def test_load_template_with_interpolation(self):
         self.run_thread('load_template_with_interpolation')
-        try:
-            self.assertEqualWithinTol(2023.46, ui.get_fit_results().parvals[0], 0.001)
-            self.assertEqualWithinTol(2743.47, ui.get_fit_results().parvals[1], 0.001)
-        except:
-            self.assertEqualWithinTol(2743.47, ui.get_fit_results().parvals[0], 0.001)
-            self.assertEqualWithinTol(2023.46, ui.get_fit_results().parvals[1], 0.001)
+        pvals = ui.get_fit_results().parvals
+        pmin = pvals[0]
+        pmax = pvals[1]
+        if pmax < pmin:
+            (pmin, pmax) = (pmax, pmin)
+
+        tol = 0.001
+        self.assertEqualWithinTol(2023.46, pmin, tol)
+        self.assertEqualWithinTol(2743.47, pmax, tol)
```

Comment:
I would contend that, aside maybe adding a couple of lines to assign the observed values before the comparison, the original code was less complex, but I don't care too much, so that's ok.